maven/test-framework/org.jboss.tools.maven.reddeer/src/org/jboss/tools/maven/reddeer/requirement/ClearRepositoryRequirement.java

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributor:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.maven.reddeer.requirement;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.eclipse.reddeer.junit.requirement.Requirement;
+import org.eclipse.reddeer.requirements.property.PropertyConfiguration;
+import org.eclipse.reddeer.swt.impl.group.DefaultGroup;
+import org.eclipse.reddeer.swt.impl.table.DefaultTable;
+import org.eclipse.reddeer.workbench.ui.dialogs.WorkbenchPreferenceDialog;
+import org.jboss.tools.maven.reddeer.maven.ui.preferences.ConfiguratorPreferencePage;
+import org.jboss.tools.maven.reddeer.requirement.ClearRepositoryRequirement.RemoveMavenRepository;
+import org.jboss.tools.maven.reddeer.wizards.ConfigureMavenRepositoriesWizard;
+
+public class ClearRepositoryRequirement extends PropertyConfiguration implements Requirement<RemoveMavenRepository> {
+
+	protected RemoveMavenRepository repo;
+	protected String url;
+
+	public @interface ToRemoveMavenRepository {
+		String url();
+
+		String ID();
+
+		boolean hasToExists();
+	}
+
+	@Retention(RetentionPolicy.RUNTIME)
+	@Target(ElementType.TYPE)
+	public @interface RemoveMavenRepository {
+		ToRemoveMavenRepository[] toRemoveRepositories() default {};
+	}
+
+	@Override
+	public void fulfill() {
+		ConfigureMavenRepositoriesWizard mr = openRepositoriesWizard();
+
+		for (ToRemoveMavenRepository r : repo.toRemoveRepositories()) {
+			String id = r.ID() + "-" + r.url();
+			if (r.hasToExists()) {
```

Comment:
This specific `if(){} else if(){}` has two branches with exactly the same code. what is the reason for that?